Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -547,20 +547,23 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
 SocketPair createPlatformConnection(unsigned options)
 {
     int sockets[2];
-    RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE, 0, sockets) != -1);
+#if OS(LINUX)
+    RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE | SOCK_CLOEXEC, 0, sockets) != -1);
```

Comment:
Why don't we just add the `SOCK_CLOEXEC` or not depending on options? instead of adding or removing it later?

Reply:
Personal preference. The code is slightly simpler this way. I could change it to not add `SOCK_CLOEXEC` if neither SetCloexecOnServer nor SetCloexecOnClient is passed if you want. (But in practice, the function is always called with SetCloexecOnServer, since there's never any valid reason for the UI process end of the socket to leak into the child process.)

Reply:
So are you OK with this?

Reply:
I still don't see why having to either set or unset the flag is simpler than just passing the right flags

Reply:
My preferred way:

```
SocketPair createPlatformConnection(unsigned options)
{
    int sockets[2];
#if OS(LINUX)
    RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE | SOCK_CLOEXEC, 0, sockets) != -1);

    if (!(options & SetCloexecOnServer))
        RELEASE_ASSERT(unsetCloseOnExec(sockets[1]));

    if (!(options & SetCloexecOnClient))
        RELEASE_ASSERT(unsetCloseOnExec(sockets[0]));
#else
    RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE, 0, sockets) != -1);

    if (options & SetCloexecOnServer)
        RELEASE_ASSERT(setCloseOnExec(sockets[1]));

    if (options & SetCloexecOnClient)
        RELEASE_ASSERT(setCloseOnExec(sockets[0]));
#endif
    SocketPair socketPair = { sockets[0], sockets[1] };
    return socketPair;
}
```

Revised per your suggestion:

```
SocketPair createPlatformConnection(unsigned options)
{
    int sockets[2];

#if OS(LINUX)
    if ((options & SetCloexecOnServer) || (options & SetCloexecOnClient))
    {
        RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE | SOCK_CLOEXEC, 0, sockets) != -1);

        if (!(options & SetCloexecOnServer))
            RELEASE_ASSERT(unsetCloseOnExec(sockets[1]));

        if (!(options & SetCloexecOnClient))
            RELEASE_ASSERT(unsetCloseOnExec(sockets[0]));

        SocketPair socketPair = { sockets[0], sockets[1] };
        return socketPair;
    }
#endif

    RELEASE_ASSERT(socketpair(AF_UNIX, SOCKET_TYPE, 0, sockets) != -1);

    if (options & SetCloexecOnServer)
        RELEASE_ASSERT(setCloseOnExec(sockets[1]));

    if (options & SetCloexecOnClient)
        RELEASE_ASSERT(setCloseOnExec(sockets[0]));

    SocketPair socketPair = { sockets[0], sockets[1] };
    return socketPair;
}
```

I think the first way is slightly simpler, but it's not important, so I'll update this pull request to do it the second way.

Reply:
Updated.